partiql-eval/src/main/kotlin/org/partiql/eval/PartiQLEngine.kt

```diff
@@ -24,10 +25,18 @@ import org.partiql.spi.catalog.Session
  */
 public interface PartiQLEngine {
 
-    public fun prepare(plan: Plan, session: Session, ctx: CompilerContext): PartiQLStatement
+    public fun prepare(plan: Plan, session: Session, ctx: Context, mode: Mode): PartiQLStatement
 
     public fun prepare(plan: Plan, session: Session): PartiQLStatement {
-        return prepare(plan, session, CompilerContext.builder().build())
+        return prepare(plan, session, Context.standard(), Mode.STRICT)
+    }
+
+    public fun prepare(plan: Plan, session: Session, ctx: Context): PartiQLStatement {
+        return prepare(plan, session, ctx, Mode.STRICT)
+    }
+
+    public fun prepare(plan: Plan, session: Session, mode: Mode): PartiQLStatement {
+        return prepare(plan, session, Context.standard(), mode)
     }
```

Comment:
no default mode and the ctx with/without are the overloads so something like

```suggestion
    public fun prepare(plan: Plan, mode: Mode): PartiQLStatement {
        return prepare(plan, mode, Context.NONE)
    }

    public fun prepare(plan: Plan, mode: Mode, ctx: Context): PartiQLStatement
```

I can follow up since this will go on the java PartiQLCompiler interface so no more kt engine thing.